server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -124,9 +136,10 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
                 .flatMap(inputStream ->
                     Mono.fromCallable(() -> ReadAheadInputStream.eager().of(inputStream).length(sizeThresholdInBytes))
                         .flatMap(readAheadInputStream -> putInCacheIfNeeded(bucketName, readAheadInputStream, blobId)
-                            .thenReturn(readAheadInputStream.in))))
+                            .thenReturn(readAheadInputStream.in)))
+                .map(this::increaseCacheMisses))
```

Comment:
why not `doOnNext`?

Reply:
I tested it:
When an error raised it should not inscrease `cacheMiss` but with `doOnNext, doOnSuccess` it still inscrease `cacheMiss`

Reply:
I found it on stackoverflow:
```
Mono::doOnNext triggers when the data is emitted successfully, which means the data is available and present.
Mono::doOnSuccess triggers when the Mono completes successfully - result is either T or null, which means the processing itself successfully finished regardless the state of data and is executed although the data are not available or present but the pipeline itself succeed.
```
[orinal](https://stackoverflow.com/questions/55976963/is-there-a-difference-between-doonsuccess-vs-doonnext-for-a-mono)